article/_posts/2022-04-05-SparseMultiChannelVAE.md

```diff
@@ -0,0 +1,102 @@
+this constraint as:
+
+$$
+\argmin_{q \in Q} \mathbb{E}_c [ \mathcal{D}_{KL}( q(\mathbf{z}|\mathbf{x}_c,\phi_c) \| p(\mathbf{z}|\mathbf{x}_1,\dots,\mathbf{x}_C,\theta) ) ]
+$$
+
+where $$Q$$ represents a family of distributions parametrized by $$\{\phi_i,\dots,\phi_C\}$$.
+
+By projecting all the channels to a shared latent representation, the channels are assumed to be independent from each
+other conditioned on the latent representation, which allows the authors to factorize the data likelihood
+$$p(\mathbf{x}|\mathbf{z})$$ and formulate sthe objective as:
+
+$$
+\mathcal{L} = \mathbb{E}_c [L_c - \mathcal{D}_{KL}( q(\mathbf{z}|\mathbf{x}_c) \| p(\mathbf{z}) )] \\
+\text{where} \quad L_c = \mathbb{E}_{q(\mathbf{z}|\mathbf{x}_c)} \left[ \sum_{i=1}^C \ln p (\mathbf{x}_i|\mathbf{z}) \right]
+$$
+
+## Variational Dropout for Sparse Representation
+The authors state that in experiments with non-sparse, i.e. standard, the lower-bound was maximum at convergence when
+the number of latent dimensions matched the number of true generative factors.
+
+Thus, they propose to use sparsity constraints, through a form of dropout, to impose a parsimonious latent
+representation that learns in an unsupervised way to only use some of the latent dimensions available. After training,
+the "effective" latent dimensions can easily be identified by using a threshold on the dropout rates of each latent
+dimension.
+
+
+# Data
+The authors first tested their method on synthetic data where they produce multiple channels with some correlations
+between them. They gain some insight into how their model works, e.g. a threshold on the dropout rates that
+discriminates useful latent dimensions from dimensions that can be pruned.
+
+In a second experiment, the authors analyzed 504 subjects from the ADNI cohort[^1], studying Alzheimer's disease using
+neuroimaging. The multi-channel data they used for each patient consisted of 3 imaging modalities (structural MRI of the
+gray matter only, functional FDG-PET and Amyloid-PET) and 6 clinical channels (age, results to mini-mental state
+examination, adas-cog, cdr, faq tests, and scholarity level) represented by continuous variables.
+
+
+# Results
+
+![](/article/images/SparseMultiChannelVAE/figure1.jpg)
+![](/article/images/SparseMultiChannelVAE/figure2.jpg)
+![](/article/images/SparseMultiChannelVAE/figure4.jpg)
+
+
+# Remarks
+- Given the formulation of the posterior in eqs. (2) and (3), it seems that they try to predict the whole latent
+presentation from each channel: $$q(\mathbf{z}|\mathbf{x}_c,\phi_c)$$. However, it is not clear to me how they aggregate
+the latent representations predicted for each channel into a single posterior w.r.t. an input $$\mathbf{x}$$.
+- I do not understand where the term $$\ln p(\mathbf{x})$$ in the left-hand side of eq. (4) comes from, and how the
+whole left-hand side is the expectation they are seeking to maximize.
```

Comment:
There is no such term in eq. 4.

Reply:
I understand the confusion I caused, but I'm referring to eq. 4 in the paper (which I did not include in the review itself since I did not understood what purpose it served). Would you suggest I get rid of the remark altogether in that case?

Reply:
I did have a look at Eq. 4 on the paper (in reality I had a look a its preceding form, which still has the term), but looks like I was not awake enough (sorry Nathan).

The term comes from applying the Bayes theorem to - ln p(z|x) = -[ ln (p(x|z) · p(z) /p(x)) =  ln (p(x|z) · p(z)) - ln p(x) ] = - [ ln p(x|z) + ln p (z) - ln p(x) ] = - ln p(x|z) - ln p (z) + ln p(x).

Reply:
You don't have to apologize! I got the part that they applied Bayes theorem. It's dumb on my part, but I just got that eq. 4 is just rearranging the equality from the previous form.

Reply:
Now that I understand better, I rephrased my explanation in the main body of the review, and I removed my remarks.

Reply:
OK, resolving.